cpp/daal/src/algorithms/kmeans/oneapi/kmeans_dense_lloyd_batch_kernel_ucapi.h

```diff
@@ -47,59 +47,56 @@ class KMeansDenseLloydBatchKernelUCAPI : public Kernel
-                                 oneapi::internal::UniversalBuffer & dataSq, oneapi::internal::UniversalBuffer & distances,
-                                 const services::Buffer<int> & assignments, const services::Buffer<algorithmFPType> & objFunction, uint32_t blockSize,
+    void computeAssignments(const services::Buffer<int> & assignments, uint32_t blockSize, uint32_t nClusters, services::Status * st);
+
+    void computePartialCandidates(const services::Buffer<int> & assignments, uint32_t blockSize, uint32_t nClusters, uint32_t reset,
+                                  services::Status * st);
+
+    void mergePartialCandidates(uint32_t nClusters, services::Status * st);
+
+    void partialReduceCentroids(const services::Buffer<algorithmFPType> & data, const services::Buffer<int> & assignments, uint32_t blockSize,
+                                uint32_t nClusters, uint32_t nFeatures, uint32_t doReset, services::Status * st);
+
+    void mergeReduceCentroids(const services::Buffer<algorithmFPType> & centroids, uint32_t nClusters, uint32_t nFeatures, services::Status * st);
+
+    void updateObjectiveFunction(const services::Buffer<int> & assignments, const services::Buffer<algorithmFPType> & objFunction, uint32_t blockSize,
                                  uint32_t nClusters, uint32_t doReset, services::Status * st);
+    void getNumEmptyClusters(uint32_t nClusters, services::Status * st);
+    void buildProgram(oneapi::internal::ClKernelFactoryIface & kernelFactory, uint32_t nClusters, daal::services::Status * st);
+    services::Status setEmptyClusters(NumericTable * const ntData, uint32_t nRows, uint32_t nClusters, uint32_t nFeatures,
+                                      services::Buffer<algorithmFPType> & outCentroids, algorithmFPType & objFuncCorrection);
+    services::Status initializeBuffers(uint32_t nClusters, uint32_t nFeatures, uint32_t blockSize);
+    services::Status getBlockSize(uint32_t nRows, uint32_t nClusters, uint32_t nFeatures, uint32_t & blockSize);
     uint32_t getCandidatePartNum(uint32_t nClusters);
     uint32_t getWorkgroupsCount(uint32_t rows);
     uint32_t getComputeSquaresWorkgroupsCount(uint32_t nFeatures);
     const char * getComputeSquaresKernelName(uint32_t nFeatures);
-    const char * getBuildOptions(uint32_t nClusters);
+    services::String getBuildOptions(uint32_t nClusters);
+
+    oneapi::internal::UniversalBuffer _dataSq;
+    oneapi::internal::UniversalBuffer _centroidsSq;
+    oneapi::internal::UniversalBuffer _distances;
+    oneapi::internal::UniversalBuffer _mindistances;
+    oneapi::internal::UniversalBuffer _candidates;
+    oneapi::internal::UniversalBuffer _candidateDistances;
+    oneapi::internal::UniversalBuffer _partialCandidates;
+    oneapi::internal::UniversalBuffer _partialCandidateDistances;
+    oneapi::internal::UniversalBuffer _partialCentroids;
+    oneapi::internal::UniversalBuffer _partialCentroidsCounters;
+    oneapi::internal::UniversalBuffer _numEmptyClusters;
```

Comment:
In future, it will be good to group a lot of fields into meaningful groups (`struct`). This is also relevant for big amount of input parameters in member functions.